docs/training.md

```diff
@@ -0,0 +1,103 @@
+# Training interface reference
+
+> [!NOTE]  
+> The training API is still experimental, and is subject to change.
+
+Cog's training API allows you to define a fine-tuning interface for an existing Cog model, so users of the model can bring their own training data to create derivative fune-tuned models. Real-world examples of this API in use include [fine-tuning SDXL with images](https://replicate.com/blog/fine-tune-sdxl) or [fine-tuning Llama 2 with structured text](https://replicate.com/blog/fine-tune-llama-2).
+
+## How it works
+
+If you've used Cog before, you've probably seen the [Predict](./python.md) class, which defines the interface for creating predictions against your model. Cog's training API works similarly: You define a Python function that describes the inputs and outputs of the training process. The inputs are things like training data, epochs, batch size, seed, etc. The output is typically a file with the fine-tuned weights.
+
+`cog.yaml`:
+
+```yaml
+build:
+  python_version: "3.10"
+train: "train.py:train"
+```
+
+`train.py`:
+
+```python
+from cog import BasePredictor, File
+import io
+
+def train(param: str) -> File:
+    return io.StringIO("hello " + param)
+```
+
+Then you can run it like this:
+
+```
+$ cog train -i param=train
+...
+
+$ cat weights
+hello train
+```
+
+## `Input(**kwargs)`
+
+Use Cog's `Input()` function to define each of the parameters in your `train()` function:
+
+```py
+from cog import Input, Path
+
+def train(
+    train_data: Path = Input(description="HTTPS URL of a file containg training data"),
+    learning_rate: float = Input(description="learning rate, for learning!", default=1e-4, ge=0),
+    seed: int = Input(description="random seed to use for training", default=42)
```

Comment:
```suggestion
    seed: int = Input(description="random seed to use for training", default=None)
```

I get that this is illustrative rather than real, but I've seen users get confused about seeds before, and we've certainly seen models with bugs related to this. If we want to use seed here, I think it'd be better to default it to `None`, as that's almost certainly the desired behaviour.